pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConsumerFactory.java

```diff
@@ -51,6 +51,7 @@ void init(StreamConfig streamConfig) {
    * @param groupId consumer group Id
    * @return the stream level consumer
    */
+  @Deprecated
   public abstract StreamLevelConsumer createStreamLevelConsumer(String clientId, String tableName,
```

Comment:
We can provide a default implementation of throwing `UnsupportedOperationException` and remove all the overrides

Reply:
Sure.